utils/data.py

```diff
@@ -12,8 +12,10 @@ def __init__(self, *args, prefix=None, **kwargs):
     async def on_message(self, msg):
         if not self.is_ready() or msg.author.bot or not permissions.can_handle(msg, "send_messages"):
             return
-
-        await self.process_commands(msg)
+        if await listener.check_for_profanity(msg):
+            await msg.channel.send("Bad word detected!")
```

Comment:
The code looks good. I do think that it might be appropriate in the response to indicate which word was bad. Not only so that we know which word it detected, but maybe also so any user would know which word to not say anymore. Really the code looks great and I'm nitpicking.

Reply:
I think you really shouldn't repeat the bad word in an open channel. Especially if it's something really bad like the n-word or other slurs. Logging it in an private admin channel would be reasonable, but out of scope for this task (suitable for #40, though).